apiserver/pkg/printers/projectcalico/printers.go

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) 2020-2021 Tigera, Inc. All rights reserved.
+// Copyright 2017 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package projectcalico
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/duration"
+	"k8s.io/kubernetes/pkg/printers"
+
+	calico "github.com/projectcalico/api/pkg/apis/projectcalico/v3"
+)
+
+// AddHandlers adds print handlers for projectcalico types.
+func AddHandlers(h printers.PrintHandler) {
```

Comment:
Will we put all resource handles in this function? Should we split this out so that each resource gets its own handler function? This might get messy if we try to put all the table printers into a single file.

Reply:
I was asking the same question to myself but decided to `copy` what kubernetes does so that it is easier for people to understand. https://github.com/kubernetes/kubernetes/blob/v1.22.1/pkg/printers/internalversion/printers.go#L89

It is a 50/50 to me, we can have one handler function for each resource and split it into different files if you prefer that way.

Reply:
huh, weird. I'm fine with leaving it as is in the PR for now then!

Reply:
I agree with you it is much easier to add wide view support if each resource gets its' own file. Considering we are adding the support to number of Calico resources, I've made the code changes accordingly.